core/src/main/java/org/infinispan/container/offheap/OffHeapDataContainer.java

```diff
@@ -509,13 +510,16 @@ public void evict(WrappedBytes key) {
       lock.lock();
       try {
          checkDeallocation();
-         InternalCacheEntry<WrappedBytes, WrappedBytes> prev = get(key);
+         long address = memoryLookup.getMemoryAddress(key);
+         InternalCacheEntry<WrappedBytes, WrappedBytes> prev = address == 0 ? null : performGet(address, key);
```

Comment:
Are you just trying to avoid the read lock acquisition here?

Reply:
I am trying to avoid any redundancy and organize publicly accessible vs. internal. Both the lock acquisition and deallocation check should be in the publicly accessible methods, and public methods should call only internal...

Reply:
No worries, I just wanted to make sure I wasn't missing something that was "fixed" and didn't quite understand.